src/evergreen/api.py

```diff
@@ -1079,6 +1079,21 @@ def get_users_for_role(self, role: str) -> UsersForRole:
         url = self._create_url(f"/roles/{role}/users")
         return UsersForRole(self._call_api(url, method="GET").json(), self)
 
+    def all_user_permissions_for_resource(
+        self, resource_id: str, resource_type: PermissionableResourceType
+    ) -> Dict[str, Dict[str, int]]:
+        """
+        Get all users with their permissions to the resource.
+
+        :param resource_id: Id of the resource to get users for.
+        :param resource_type: Resource type of the resource.
+        :return: A dict containing user: permissions mappings.
+        """
+        url = self._create_url("/users/permissions")
+        return self._call_api(
+            url, data=json.dumps({"resource_id": resource_id, "resource_type": resource_type})
+        ).json()
```

Comment:
Could you not return the raw json here, but wrap it in a model object like we do for all the other endpoints?

Reply:
Hi @dbradf. I was going to respond saying that the returned response contains variable keys making the modeling somewhat difficult. But I took a stab at it anyway, would like to know what you think.